### PR TITLE
[4.0] Remove the deprecated UserWrapper class

### PR DESCRIPTION
**Pull request by laoneo**
Removes the deprecated `Joomla\CMS\User\UserWrapper` class.

### CODE REVIEW

File: libraries/src/User/User.php [hunk 1]
```diff
@@ -648,14 +625,14 @@ public function bind(&$array)
...
-				if ($this->requireReset == 1 && $this->userHelper->verifyPassword($this->password_clear, $this->password))
+				if ($this->requireReset == 1 && UserHelper::verifyPassword($this->password_clear, $this->password))
 				{
 					$this->setError(Text::_('JLIB_USER_ERROR_CANNOT_REUSE_PASSWORD'));
 
 					return false;
 				}
 
-				$array['password'] = $this->userHelper->hashPassword($array['password']);
+				$array['password'] = UserHelper::hashPassword($array['password']);
```
Comment (wilsonge) [inline libraries/src/User/User.php:635]: can we use `static` rather than `UserHelper` please (i.e. use late static bindings)

Comment (laoneo) [reply libraries/src/User/User.php:635]: We are not in the UserHelper class.

Comment (wilsonge) [reply libraries/src/User/User.php:635]: *facepalm*

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**